pkg/disks/loop_test.go

```diff
@@ -0,0 +1,700 @@
+			}
+
+			symlinksDir := path.Join(devfsDir, "loops")
+			symlinksDirStat, err := os.Stat(symlinksDir)
+			if err != nil && !errors.Is(err, fs.ErrNotExist) {
+				t.Fatalf("unexpected error on symlinksDir %q stat: %v", symlinksDir, err)
+			}
+
+			symlinksDirExists := symlinksDirStat != nil
+			if !reflect.DeepEqual(tc.expectedDevLoopsDirToExist, symlinksDirExists) {
+				t.Fatalf("expected symlinksDir to exists %v, got %v", tc.expectedDevLoopsDirToExist, symlinksDirExists)
+			}
+
+			if symlinksDirExists {
+				var existingLoopDevicesSymlinks []deviceSymlink
+				err = filepath.WalkDir(symlinksDir, func(fpath string, d fs.DirEntry, err error) error {
+					if err != nil {
+						return err
+					}
+
+					if fpath == symlinksDir {
+						return nil
+					}
+
+					if d.Type()&fs.ModeSymlink == 0 {
+						return fmt.Errorf("expected to find only symlinks in symlinks directory, got %v filemode", d.Type())
+					}
+
+					device, err := os.Readlink(fpath)
+					if err != nil {
+						return fmt.Errorf("can't read symlink %q: %w", fpath, err)
+					}
+
+					existingLoopDevicesSymlinks = append(existingLoopDevicesSymlinks, deviceSymlink{
+						symlinkPath: fpath,
+						devicePath:  device,
+					})
+
+					return nil
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				expectedDeviceSymlinks := tc.expectedDeviceSymlinks(devfsDir)
+
+				if !reflect.DeepEqual(expectedDeviceSymlinks, existingLoopDevicesSymlinks) {
+					t.Fatalf("expected device symlinks %v, got %v", expectedDeviceSymlinks, existingLoopDevicesSymlinks)
+				}
+			}
+		})
+	}
+}
+
+func TestDeleteLoopDevice(t *testing.T) {
+	t.Parallel()
+
+	type deviceSymlink struct {
+		symlinkPath string
+		devicePath  string
+	}
+
+	tt := []struct {
+		name                       string
+		deviceSymlinkPath          func(devfsPath string) string
+		devicePath                 func(devfsPath string) string
+		imagePath                  func(tmpDir string) string
+		populateFilesystem         func(tmpDir, devfsPath string) error
+		expectedCommands           func(tmpDir, devfsPath string) []exectest.Command
+		expectedDevLoopsDirToExist bool
+		expectedDeviceSymlinks     func(devfsPath string) []deviceSymlink
+		expectedErr                func(tmpDir, devfsPath string) error
+	}{
+		{
+			name: "detaches loop device, removes image and symlink together with symlinks directory when it's the last device",
+			deviceSymlinkPath: func(devfsPath string) string {
+				return path.Join(devfsPath, "loops", "foo")
+			},
+			devicePath: func(devfsPath string) string {
+				return path.Join(devfsPath, "loop0")
+			},
+			imagePath: func(tmpDir string) string {
+				return path.Join(tmpDir, "image")
+			},
+			populateFilesystem: func(tmpDir, devfsPath string) error {
+				imagePath := path.Join(tmpDir, "image")
+				f, err := os.Create(imagePath)
+				if err != nil {
+					return fmt.Errorf("can't create image file at %q: %w", imagePath, err)
```

Comment:
i think these could just call t.Fail right here for better context

Reply:
done